### PR TITLE
Add Felucia

### PR DESCRIPTION
**Pull request by WarriorGallade**

### CODE REVIEW

File: data/mods/gen9ssb/random-teams.ts [hunk 1]
```diff
@@ -1009,10 +1016,17 @@ export class RandomStaffBrosTeams extends RandomTeams {
...
 			while (moves.length < 3 && ssbSet.moves.length > 0) {
 				let move = this.sampleNoReplace(ssbSet.moves);
 				if (Array.isArray(move)) move = this.sampleNoReplace(move);
+				if (ssbSet.species === "Vespiquen") {
+					if (move === 'Calm Mind' && moves.includes('Night Shade')) {
+						move = this.sample(['Thief', 'Toxic']);
+					} else if (move === 'Night Shade' && moves.includes('Calm Mind')) {
+						move = 'Bug Buzz';
+					}
+				}
```
Comment (HisuianZoroark) [inline data/mods/gen9ssb/random-teams.ts:1029]:
```suggestion
				if (ssbSet.species === "Vespiquen") {
					if (move === 'Calm Mind' && moves.includes('Night Shade')) {
						move = this.sample(['Thief', 'Toxic']);
					} else if (move === 'Night Shade' && moves.includes('Calm Mind')) {
						move = 'Bug Buzz';
					}
				}
```
There is a dedicated section for set specific tweaks down further, migrate the code there

File: server/chat-plugins/randombattles/index.ts [hunk 2]
```diff
@@ -424,6 +424,7 @@ function generateSSBSet(set: SSBSet, dex: ModdedDex, baseDex: ModdedDex) {
 	buf += `<ul style="list-style-type:none;"><li>${set.species}${set.gender !== '' ? ` (${set.gender})` : ``}`;
 	buf += `${set.item ? ' @ ' : ''}${Array.isArray(set.item) ? set.item.map(x => dex.items.get(x).name).join(' / ') : dex.items.get(set.item).name}</li>`;
 	buf += `<li>Ability: ${Array.isArray(set.ability) ? set.ability.map(x => dex.abilities.get(x).name).join(' / ') : dex.abilities.get(set.ability).name}</li>`;
+	// eslint-disable-next-line max-len
 	if (set.teraType) buf += `<li>Tera Type: ${Array.isArray(set.teraType) ? set.teraType.map(x => dex.types.get(x).name).join(' / ') : set.teraType === 'Any' ? 'Any' : dex.types.get(set.teraType).name}</li>`;
```
Comment (KrisXV) [inline server/chat-plugins/randombattles/index.ts:428]:
```suggestion
	if (set.teraType){
		buf += `<li>Tera Type: ${Array.isArray(set.teraType) ? set.teraType.map(x => dex.types.get(x).name).join(' / ') : set.teraType === 'Any' ? 'Any' : dex.types.get(set.teraType).name}</li>`;
	}
```

File: data/mods/gen9ssb/moves.ts [hunk 3]
```diff
@@ -1654,6 +1654,42 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		type: "Grass",
 	},
 
+	// Felucia
+	riggeddice: {
+		accuracy: 100,
+		basePower: 0,
+		category: "Status",
+		shortDesc: "Inverts target's stat boosts if they have any, taunts otherwise. User then switches.",
```
Comment (KrisXV) [inline data/mods/gen9ssb/moves.ts:1662]:
```suggestion
		shortDesc: "Stat changes: inverts, else Taunt. User switches.",
```

File: data/mods/gen9ssb/conditions.ts [hunk 4]
```diff
@@ -667,6 +667,18 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
...
 		},
 	},
+	felucia: {
+		noCopy: true,
+		onStart() {
+			this.add(`c:|${getName('Felucia')}|Good morning gamers! Just here to fix a few things`);
+		},
+		onSwitchOut() {
+			this.add(`c:|${getName('Felucia')}|I have bots to make and chatrooms to manage...`);
+		},
```
Comment (KrisXV) [inline data/mods/gen9ssb/conditions.ts:677]:
```suggestion
		onSwitchOut(pokemon) {
			this.add(`c:|${getName('Felucia')}|I have bots to make and chatrooms to manage...`);
			if (pokemon.illusion) return;
			pokemon.heal(pokemon.baseMaxhp / 3);
		},
```

File: data/mods/gen9ssb/conditions.ts [hunk 5]
```diff
@@ -667,6 +667,18 @@ export const Conditions: {[k: string]: ModdedConditionData & {innateName?: strin
...
+		onFaint(pokemon) {
+			this.add(`c:|${getName('Felucia')}|Okay that's enough work for today`);
+		},
```
Comment (KrisXV) [inline data/mods/gen9ssb/conditions.ts:680]:
```suggestion
		},
		innateName: "Regenerator",
```

File: server/chat-plugins/randombattles/index.ts [hunk 6]
```diff
@@ -424,7 +424,9 @@ function generateSSBSet(set: SSBSet, dex: ModdedDex, baseDex: ModdedDex) {
 	buf += `<ul style="list-style-type:none;"><li>${set.species}${set.gender !== '' ? ` (${set.gender})` : ``}`;
 	buf += `${set.item ? ' @ ' : ''}${Array.isArray(set.item) ? set.item.map(x => dex.items.get(x).name).join(' / ') : dex.items.get(set.item).name}</li>`;
 	buf += `<li>Ability: ${Array.isArray(set.ability) ? set.ability.map(x => dex.abilities.get(x).name).join(' / ') : dex.abilities.get(set.ability).name}</li>`;
-	if (set.teraType) buf += `<li>Tera Type: ${Array.isArray(set.teraType) ? set.teraType.map(x => dex.types.get(x).name).join(' / ') : set.teraType === 'Any' ? 'Any' : dex.types.get(set.teraType).name}</li>`;
+	if (set.teraType){
```
Comment (KrisXV) [inline server/chat-plugins/randombattles/index.ts:427]:
```suggestion
	if (set.teraType) {
```

**Pull request merged**